Set correct entrypoint to module in package.json

schwarmco opened the pull request:
Fixes #1560
Fixes #1553

cristiand391 commented on package.json:
```diff
@@ -23,7 +23,7 @@
   ],
   "main": "./index",
-  "module": "./module/index",
+  "module": "./index",
```
we don't ship an ESM build yet so it's better to just remove this field (`module` in pjson make tools think this is the ESM entrypoint, we don't want CJS be treated as ESM).

cristiand391 reviewed (changes requested):
hey @schwarmco, sorry for the late reply.

jsforce v3 doesn't ship an ESM build yet so it will be better to just remove the `module` field for now.

schwarmco commented:
> jsforce v3 doesn't ship an ESM build yet so it will be better to just remove the `module` field for now.

that makes perfect sense. can confirm it still works with the import-syntax.
i've rebased and removed the line from package.json

cristiand391 commented:
we got 1 external NUT failure but it's unrelated to this change, merging this...

thanks for the PR @schwarmco :)
